### PR TITLE
Fix #1673, fix #1664, update adblock to 4.x

### LINKED ISSUE #1664: Update ad block to 4.x
https://github.com/brave/ad-block

Test plan:
Test against popular websites to see if it's blocking ads correctly.
Testing non-english regional filter ads would be nice too, compare old and new builds in terms of blocking quality.

### LINKED ISSUE #1673: Adblock regional filter network request called on every preference change.

### PR DESCRIPTION
**Pull request by iccub**
## Changes
- `ad-block` upgraded to `4.1.1`
- `tracking-protection` upgraded to `1.2.0`
- ad block data file version to `4`

Not much changed on iOS side of things. Updated script for getting regional add block lists, changed data version to 4, and had to add header search paths for ad-block headers.

It would be nice if someone could try to build it on their machine to verify that it works well.

### CODE REVIEW

File: brave/src/webfilters/AdBlocker.swift [hunk 1]
```diff
@@ -16,7 +16,7 @@ class AdBlocker {
     static let prefKeyDefaultValue = true
     static let prefKeyUseRegional = "braveAdblockUseRegional"
     static let prefKeyUseRegionalDefaultValue = true
-    static let dataVersion = "3"
+    static let dataVersion = "4"
```
Comment (bbondy) [inline brave/src/webfilters/AdBlocker.swift]: 👍

**Comment by iccub** [comment]
Added:
- removing old `.dat` files after new data version is available.
- update url for adblock data
- remove amount of calls to get adblock data

File: brave/src/webfilters/AdBlocker.swift [hunk 2]
```diff
@@ -4,70 +4,114 @@ import Foundation
...
+    }
+    
+    /// We want to avoid situations in which user still has downloaded old abp data version.
+    /// We remove all abp data after data version is updated, then the newest data is downloaded. 
+    private func setDataVersionPreference() {
+        guard let prefs = BraveApp.getPrefs() else {
+            log.error("No prefs found")
+            return
+        }
+        guard let dataVersionPref = prefs.intForKey(AdBlocker.dataVersionPrefKey), dataVersionPref == AdBlocker.dataVersion else {
```
Comment (jhreis) [inline brave/src/webfilters/AdBlocker.swift]: Clean solution, also will clear out files if there is no preference set.

**Comment by iccub** [comment]
Updated ad block version to this commit https://github.com/brave/ad-block/commit/b933cf5168c5284dc159735d8bcefb415d0cfc00

We have to use the commit hash dependency until a new adblock npm version release.

**Pull request merged**